pkg/controllers/drift/controller.go

```diff
@@ -0,0 +1,137 @@
+package drift
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/karpenter-core/pkg/apis/provisioning/v1alpha5"
+	operatorcontroller "github.com/aws/karpenter-core/pkg/operator/controller"
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+	"github.com/aws/karpenter/pkg/cloudprovider"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/pkg/logging"
+	"net/http"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+	"time"
+)
+
+const controllerName = "drift"
+
+type Drift struct {
+	kubeClient    client.Client
+	cloudProvider *cloudprovider.CloudProvider
+}
+
+// NewController constructs a controller instance
+func NewController(kubeClient client.Client, cloudProvider *cloudprovider.CloudProvider) *Drift {
```

Comment:
There's the introduction of the `controller.Typed` into the `karpenter-core` project which means that you can create this controller as a `controller.Typed[*v1.Node]()` with automatic getting and patching.

Here's an example since we're trying to migrate all controllers over to the framework if we can:

https://github.com/aws/karpenter-core/blob/main/pkg/controllers/node/controller.go#L58

Reply:
Done.